app/code/Magento/QuoteGraphQl/Model/Resolver/AssignCustomerToGuestCart.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Resolver;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\StateException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\GraphQl\Model\Query\ContextInterface;
+use Magento\Quote\Api\GuestCartManagementInterface;
+use Magento\Quote\Model\Cart\CustomerCartResolver;
+use Magento\QuoteGraphQl\Model\Cart\GetCartForUser;
+
+/**
+ * Assign the customer to the guest cart resolver
+ *
+ * @SuppressWarnings(PHPMD.LongVariable)
+ */
+class AssignCustomerToGuestCart implements ResolverInterface
+{
+    /**
+     * @var GetCartForUser
+     */
+    private $getCartForUser;
+
+    /**
+     * @var CustomerCartResolver
+     */
+    private $customerCartResolver;
+
+    /**
+     * @var GuestCartManagementInterface
+     */
+    private $guestCartManagementInterface;
+
+    /**
+     * @param GetCartForUser $getCartForUser
+     * @param GuestCartManagementInterface $guestCartManagementInterface
+     * @param CustomerCartResolver $customerCartResolver
+     */
+    public function __construct(
+        GetCartForUser $getCartForUser,
+        GuestCartManagementInterface $guestCartManagementInterface,
+        CustomerCartResolver $customerCartResolver
+    ) {
+        $this->getCartForUser = $getCartForUser;
+        $this->guestCartManagementInterface = $guestCartManagementInterface;
+        $this->customerCartResolver = $customerCartResolver;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        /** @var ContextInterface $context */
+        if (false === $context->getExtensionAttributes()->getIsCustomer()) {
+            throw new GraphQlAuthorizationException(__(
+                'The current customer isn\'t authorized.'
+            ));
+        }
+
+        $currentUserId = $context->getUserId();
+        $guestMaskedCartId = $args['cart_id'];
```

Comment:
Hello @khrystynastolbova,
thank you for your contribution

could you add a check on cart_id?

 if (empty($args['cart_id'])) {
            throw new GraphQlInputException(__('Required parameter "cart_id" is missing'));
        }

Reply:
Hi @danielrussob, actually `cart_id` is a required field at the schema level, and personally don't see the reason to have an additional check for that.